### PR TITLE
Get all targets building off of updated base/net

### PR DESCRIPTION
**Pull request by andrewsavage1**

### CODE REVIEW

File: third_party/ots/src/graphite.h [hunk 1]
```diff
@@ -14,6 +14,7 @@ template<typename ParentType>
 class TablePart {
  public:
   TablePart(ParentType* parent) : parent(parent) { }
+  virtual ~TablePart() { }
```
Comment (dahlstrom-g) [inline third_party/ots/src/graphite.h:17]: What's this for?

File: third_party/v8/src/base/logging.h [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_LOGGING_H_
 #define V8_BASE_LOGGING_H_
 
+#include <cstdint>
```
Comment (dahlstrom-g) [inline third_party/v8/src/base/logging.h:8]: What's this for?

File: third_party/v8/src/base/macros.h [hunk 3]
```diff
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_MACROS_H_
 #define V8_BASE_MACROS_H_
 
+#include <cstdint>
```
Comment (dahlstrom-g) [inline third_party/v8/src/base/macros.h:8]: [ditto]

File: third_party/v8/src/compiler/bytecode-analysis.cc [hunk 4]
```diff
@@ -225,7 +225,7 @@ void UpdateOutLiveness(Bytecode bytecode, BytecodeLivenessState* out_liveness,
     int target_offset = accessor.GetJumpTargetOffset();
     out_liveness->Union(*liveness_map.GetInLiveness(target_offset));
   } else if (Bytecodes::IsSwitch(bytecode)) {
-    for (const auto& entry : accessor.GetJumpTableTargetOffsets()) {
+    for (auto entry : accessor.GetJumpTableTargetOffsets()) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/bytecode-analysis.cc:228]: What's this for?

File: third_party/v8/src/compiler/bytecode-analysis.cc [hunk 5]
```diff
@@ -521,7 +521,7 @@ void BytecodeAnalysis::Analyze() {
         liveness_map_.GetLiveness(current_offset);
 
     bool any_changed = false;
-    for (const auto& entry : iterator.GetJumpTableTargetOffsets()) {
+    for (auto entry : iterator.GetJumpTableTargetOffsets()) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/bytecode-analysis.cc:524]: [ditto]

File: third_party/v8/src/compiler/bytecode-graph-builder.cc [hunk 6]
```diff
@@ -3574,7 +3574,7 @@ void BytecodeGraphBuilder::BuildSwitchOnSmi(Node* condition) {
       bytecode_iterator().GetJumpTableTargetOffsets();
 
   NewSwitch(condition, offsets.size() + 1);
-  for (const auto& entry : offsets) {
+  for (auto entry : offsets) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/bytecode-graph-builder.cc:3577]: [ditto]

File: third_party/v8/src/compiler/pipeline.cc [hunk 7]
```diff
@@ -811,7 +811,7 @@ void PrintFunctionSource(OptimizedCompilationInfo* info, Isolate* isolate,
         int len = shared->EndPosition() - start;
         SubStringRange source(String::cast(script->source()), no_allocation,
                               start, len);
-        for (const auto& c : source) {
+        for (auto c : source) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/pipeline.cc:814]: [ditto]

File: third_party/v8/src/compiler/serializer-for-background-compilation.cc [hunk 8]
```diff
@@ -2770,7 +2770,7 @@ void SerializerForBackgroundCompilation::VisitSwitchOnSmiNoFeedback(
     interpreter::BytecodeArrayIterator* iterator) {
   interpreter::JumpTableTargetOffsets targets =
       iterator->GetJumpTableTargetOffsets();
-  for (const auto& target : targets) {
+  for (auto target : targets) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/serializer-for-background-compilation.cc:2773]: [ditto]

File: third_party/v8/src/compiler/store-store-elimination.cc [hunk 9]
```diff
@@ -448,7 +448,7 @@ UnobservablesSet UnobservablesSet::Intersect(const UnobservablesSet& other,
   if (IsEmpty() || other.IsEmpty()) return empty;
 
   UnobservablesSet::SetT* intersection = NewSet(zone);
-  for (const auto& triple : set()->Zip(*other.set())) {
+  for (auto triple : set()->Zip(*other.set())) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/store-store-elimination.cc:451]: [ditto]

File: third_party/v8/src/compiler/store-store-elimination.cc [hunk 10]
```diff
@@ -474,7 +474,7 @@ UnobservablesSet UnobservablesSet::RemoveSameOffset(StoreOffset offset,
   *new_set = *set();
 
   // Remove elements with the given offset.
-  for (const auto& entry : *new_set) {
+  for (auto entry : *new_set) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/compiler/store-store-elimination.cc:477]: [ditto]

File: third_party/v8/src/inspector/v8-string-conversions.h [hunk 11]
```diff
@@ -6,6 +6,7 @@
 #define V8_INSPECTOR_V8_STRING_CONVERSIONS_H_
 
 #include <string>
+#include <cstdint>
```
Comment (dahlstrom-g) [inline third_party/v8/src/inspector/v8-string-conversions.h:9]: [ditto]

File: third_party/v8/src/interpreter/bytecode-array-accessor.cc [hunk 12]
```diff
@@ -345,7 +345,7 @@ JumpTableTargetOffsets::iterator JumpTableTargetOffsets::end() const {
 int JumpTableTargetOffsets::size() const {
   int ret = 0;
   // TODO(leszeks): Is there a more efficient way of doing this than iterating?
-  for (const auto& entry : *this) {
+  for (auto entry : *this) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/interpreter/bytecode-array-accessor.cc:348]: [ditto]

File: third_party/v8/src/objects/code.cc [hunk 13]
```diff
@@ -849,7 +849,7 @@ void BytecodeArray::Disassemble(std::ostream& os) {
     if (interpreter::Bytecodes::IsSwitch(iterator.current_bytecode())) {
       os << " {";
       bool first_entry = true;
-      for (const auto& entry : iterator.GetJumpTableTargetOffsets()) {
+      for (auto entry : iterator.GetJumpTableTargetOffsets()) {
```
Comment (dahlstrom-g) [inline third_party/v8/src/objects/code.cc:852]: [ditto]

File: third_party/v8/src/base/logging.h (hunk 2 above)
Comment (andrewsavage1) [reply third_party/v8/src/base/logging.h:8]: Due to changes in inlcudes in base/ files, includes had to be added where they were actually needed

File: third_party/v8/src/compiler/bytecode-analysis.cc (hunk 4 above)
Comment (andrewsavage1) [reply third_party/v8/src/compiler/bytecode-analysis.cc:228]: base iterator typing was changed, so how we accessed it needed to change. I believe this is the same thing that's happening upstream now

File: cobalt/network/net_poster.cc [hunk 14]
```diff
@@ -59,27 +59,30 @@ void NetPoster::Send(const GURL& url, const std::string& content_type,
...
-                   [&](const std::unique_ptr<net::URLFetcher>& i) {
-                     return source == i.get();
-                   });
-  if (it != fetchers_.end()) {
-    fetchers_.erase(it);
-  } else {
-    NOTREACHED() << "Expected to find net::URLFetcher";
-  }
-}
+void NetPoster::OnReadCompleted(net::URLRequest* request, int bytes_read) {}
```
Comment (andrewsavage1) [inline cobalt/network/net_poster.cc:68]: Looks iffy

File: cobalt/network/network_module.cc [hunk 15]
```diff
@@ -209,18 +209,18 @@ void NetworkModule::OnCreate(base::WaitableEvent* creation_event) {
 
   net::NetLog* net_log = NULL;
 #if defined(ENABLE_NETWORK_LOGGING)
-  net_log = net_log_.get();
+  // net_log = net_log_.get();
```
Comment (andrewsavage1) [inline cobalt/network/network_module.cc]: Investigate

**Pull request merged**